Initial SECURITY.md (#1)
alphagov/.github: SECURITY.md

# Security Notice

## What?

> This is the security notice for all GDS (Government Digital Service) `alphagov` repositories. The notice explains how vulnerabilities should be reported to GDS. At GDS there are cyber security and information assurance teams, as well as security-conscious people within the programmes, that assess and triage all reported vulnerabilities.

Reviewer: Don't we have other GitHub organisations besides `alphagov`? i.e. https://github.com/openregister

Reviewer 2: Good point. And https://github.com/gds-operations. @deanwilson you recently went and found all these. Are there any others?

Reviewer 3: ~~There's also https://github.com/openregister or similar which GDS owns.~~ that's what happens when you only read the github notification.

Author: Thanks all - removed `alphagov`.

## Reporting a Vulnerability

> GDS is an advocate of responsible vulnerability disclosure. If you’ve found a vulnerability, great! We would like to know so we can fix it.

Reviewer 2: I don't like the "great!" bit. It's really not great at all. I think just join the two sentences:

> If you’ve found a vulnerability, we would like to know so we can fix it.

Reviewer 4: I guess it's not great that the code has a vulnerability, but it is great that someone found it and wants to report it to us. For clarity though, I agree, we should remove the `great!`.

If you believe a vulnerability affects the data (release of sensitive data or personal identifiable information) of a system, send details to [disclosure@digital.cabinet-office.gov.uk], include:
- the website, page or repository where the vulnerability can be observed

> - a brief description of the type of vulnerability and any related [OWASP category]

Reviewer: Isn't the bare minimum we require is a description? 
```suggestion
- a brief description of the vulnerability
```
I feel like asking the person to try and classify the vulnerability is increases the barrier to them actually reporting it.

Author: I've separated the two points and said the type is optional - I think this is useful information in what the researcher/bug finder believes the vulnerability is.

- non-destructive exploitation details

> For all other vulnerabilities, you can leave an issue against the repository (if you are not sure about which repository to raise an issue against, you can still email details through to [disclosure@digital.cabinet-office.gov.uk]):

Reviewer 2: As discussed in Slack, I don't like having two different approaches. Copy-and-pasted-ish from Slack:

* I don't think the "leave an issue against the repo" approach will work for GOV.UK. We have >200 repos, and the way those repos fit together to create GOV.UK's various user interfaces is opaque to external folks (and many internal folks, likely). So someone finding a potential vulnerability in GOV.UK would just be guessing at which repo to report the issue on. In addition, we're not monitoring issues across all repos, because there are simply too many (repos). Having to either add processes or tooling to do this feels unnecessary, when we already have the email triage point.

* I don't like the idea of public disclosure of a potential vulnerability until a fix is in place, even if the details of the vulnerability aren't included. Having someone say "I've found an XSS on your site" could just act like a red flag for other people to start attacking the system. Also, having a publicly-viewable channel like this still has the risk that someone will disclose the vulnerability itself (rather than just the existence of a possible vulnerability).

* I assume the reason for having two separate entry points is to limit the impact on Cyber in terms of triaging all submissions? Given the fact that we've already acknowledged that it's very difficult for external users to know who to report an issue to, having them self-select a repo feels like we'll just end up relying on individual teams to bounce things around saying "got this, not ours, whose is it?"

* I'd much rather that someone central triaged things, and could quickly tell who to pass them on to (whether internally to GDS, or out to NCSC or another department). Can we not start with the single email route, and then reconsider if it's found to be overwhelming / not working as expected? Start simple, and iterate based on data.

Author: Agreed, thanks Tim.

- prefix the issue title with `[Vulnerability]`
- add details to the issue about the impact and any related [OWASP category]
- don’t include detailed exploitation steps publically, GDS may respond in the issue's comments or contact you via your GitHub profile (alternatively, you can leave other contact means)

> GDS use ZenDesk as a ticketing system, so you may get a reply or response from that platform.

Reviewer: A lot of people are unfamiliar with ZenDesk - isn't it just an email to the end user?

Author: The email address goes to a ZenDesk queue, so I think it's worth mentioning that they might get a reply from that platform

## Scope
The following are **not** in scope:

> - volumetric vulnerabilities - meaning no overwhelming a service with a high volume of requests 

Reviewer: This felt difficult to read
```suggestion
- volumetric vulnerabilities, for example overwhelming a service with a high volume of requests 
```

Author: Fixed - thanks!

# Security Notice

## What?

This is the security notice for all Government Digital Service (GDS) `alphagov` repositories. The notice explains how vulnerabilities should be reported to GDS. At GDS there are cyber security and information assurance teams, as well as security-conscious people within the programmes, that assess and triage all reported vulnerabilities.

## Reporting a Vulnerability

> GDS are advocates of responsible vulnerability disclosure. If you’ve found a vulnerability, great! We would like to know so we can fix it.

Reviewer 5: GDS is an advocate

- If you believe a vulnerability affects the data (release of sensitive data or personal identifiable information) of a system: 
  - Send details to [disclosure@digital.cabinet-office.gov.uk], include:
    - the website, page or repository where the vulnerability can be observed
    - a brief description of the type of vulnerability and any related [OWASP category]
    - non-destructive exploitation details
  - Prefix your subject with `[Vulnerability]`
  - GDS use ZenDesk as a ticketing system, so you may get a reply or response from that platform

- For all other vulnerabilities, you can leave an issue against the repository:
  - (if you are not sure about which repository to raise an issue against, you can still email details through to [disclosure@digital.cabinet-office.gov.uk])
  - Prefix the issue title with `[Vulnerability]`
  - Label the issue `vulnerability`

>   - Add details to the issue about the impact and any related [OWASP category]

Reviewer 5: link to the categories

Author: Is already :)

  - Don’t include detailed exploitation steps publically, GDS may respond in the issue's comments or contact you via your GitHub profile (alternatively, you can leave other contact means)

## Scope
The following are **not** in scope:
- volumetric vulnerabilities - meaning no overwhelming a service with a high volume of requests 
- reports indicating that our services do not fully align with “best practice”, for example missing security headers

If you are not sure, you can still reach out via email; [disclosure@digital.cabinet-office.gov.uk].

## Bug bounty

> Unfortunately, GDS doesn't offer a paid bug bounty programme. GDS will make efforts to show appreciation to people who take the time and effort to disclose vulnerabilities responsibly.

Reviewer 5: It's insane that we don't.

Reviewer 5: Can we have a whip-around? I will personally pay 100 pounds to someone who finds a problem that would have worken me up out of hours :laughing:
